packages/@okta/vuepress-site/use_cases/inline_hooks/registration_hook/registration_hook/index.md

```diff
@@ -0,0 +1,244 @@
+
+- JSON objects you can include in your response
+
+This information is specific to the Registration Inline Hook, one type of inline hook supported by Okta.
+
+## See Also
+
+For a general introduction to Okta inline hooks, see [Inline Hooks](/use_cases/inline_hooks/).
+
+For information on the API for registering external service endpoints with Okta, see [Inline Hooks Management API](/docs/api/resources/inline-hooks).
+
+For steps to enable this inline hook, see below, [Enabling a Registration Inline Hook](#enabling-a-registration-inline-hook-for-self-service-registration).
+
+## About
+
+The Okta Registration Inline Hook allows you to integrate your own custom code into Okta’s [Self-Service Registration](https://help.okta.com/en/prod/Content/Topics/Directory/eu-self-service.htm) flow. The hook is triggered after Okta receives the registration request but before the user is created. Your custom code can:
+
+- Set or override the values that will be populated in attributes of the user's Okta profile
+- Allow or deny the registration attempt, based on your own validation of the information the user has submitted
+
+## Objects in the Request from Okta
+
+The outbound call from Okta to your external service includes the following objects in its JSON payload:
+
+### data.userProfile
+
+This object contains name-value pairs for each attribute supplied by the user in the Self-Service Registration form except for password.
+
+Using the `com.okta.user.profile.update` commands you send in your response, you can modify the values of the attributes, or add other attributes, before the values are assigned to the Okta user profile that will be created for the registering user.
+
+You can only set values for profile fields which already exist in your Okta user profile schema. Registration Inline Hook functionality can only set values, it cannot create new fields.
+
+> Note: The `password` field, along with any attributes that are marked as sensitive in your Okta user schema, is omitted from the information sent to your external service in the `data.userProfile` object. The password (or even its hash) is never sent to your external service in any way.
+
+### data.action
+
+The action that Okta is currently set to take, regarding whether to allow this registration attempt.
+
+There are two possible values:
+
+- `ALLOW` indicates that the registration attempt will be allowed to proceed
+- `DENY` indicates that the registration attempt will be terminated (no user will be created in Okta)
+
+The action is `ALLOW` by default (in practice, `DENY` will never be sent to your external service).
+
+Using the `com.okta.action.update` [command](#supported-commands) in your response, you can change the action that Okta will take.
+
+## Objects in Response You Send
+
+For the Registration Inline Hook, the `commands` and `error` objects that you can return in the JSON payload of your response are defined as follows:
```

Comment:
Given that this is the first explicit mention of these objects in this doc, I would probably rephrase this to make it more explicit that these are the only fields allowed in the response (assuming we're ignoring `debugContext`). E.g.: "As with other Inline Hooks, the Registration Inline Hook response may include an array of `commands` to be executed by Okta or an `error` object to indicate problems with the registration request. These objects are defined as follows:"

Reply:
Changed to: "The objects that you can return in the JSON payload of your response are an array of one or more `commands`, to be executed by Okta, or an `error`, object to indicate problems with the registration request. These objects are defined as follows:"